### ISSUE TITLE
Forge pauses processing when the http://127.0.0.1:7860/ web browser is unfocussed, minimized, or MS Windows locked.

### ISSUE THREAD

**Issue by cokimax2** [lllyasviel/stable-diffusion-webui-forge#1927]
Forge is working good on everything now for me except this issue and "Generate forever/Cancel generate forever" not properly working.

**Symptom**

- On SD and XL model, Forge in Generate forever pauses processing when the http://127.0.0.1:7860/ web browser is minimized, or MS Windows locked.
- On Flux model,  Forge in Generate forever pauses processing when the http://127.0.0.1:7860/ web browser is unfocussed, minimized, or MS Windows locked.

**Comment by wuziq** [comment]
Noticed the same with flux.  Using Brave browser.  Need to disable screen from locking when screen saver turns on.

**Comment by RalFingerLP** [comment]
Glad I did a search before posting, I have the same issue (also Brave). It only continues to work when I actually have the browser active, or hover over the tab on my taskbar to see the little window preview, then it keeps on generating.

**Comment by Blueblizard** [comment]
> * On Flux model,  Forge in Generate forever pauses processing when the http://127.0.0.1:7860/ web browser is unfocussed, minimized, or MS Windows locked.

Yes, i didn´t test in XL but i have the same problem in FLUX.

**Comment by timwillhack** [comment]
Same problem in flux, fully updated

**Comment by webhappy99** [comment]
I'm having the same issue with Opera browser and SDXL.

**Comment by johnnykorm** [comment]
Same problem with Chrome browser and Flux.

**Comment by PixelClassicist** [comment]
Same here

**Comment by Cranbert** [comment]
Have this issue, _and_ also Brave uses way more CPU when the Forge tab is focused, which makes switch tabs necessary.

**Comment by Delorean-F** [comment]
I have the same problem.
If I minimize my browser window or open other tabs during “Generate Forever”, the generation of the next image never starts and only starts when I open the WebUI Forge window again.

**Comment by coghazard** [comment]
Same issue on Firefox ESR and SDXL, fully updated.

**Comment by suminexrepo** [comment]
It's not a minor problem.. This is a VERY annoying issue with a staple feature of the a1111 webui. Very frustrating.

**Comment by Blueblizard** [comment]
> It's not a minor problem.. This is a VERY annoying issue with a staple feature of the a1111 webui. Very frustrating.

Yes, it´s a very important issue and it's been more than a month since this happened.. I'm very frustrated too.

**Comment by RalFingerLP** [comment]
I found a workaround to fix the issue by accident! Drag the UI Window behind your TaskBar in Windows (i am running Windows10) so it overlaps behind it like out of your screen at the bottom, and it works! Not sure what is the cause for this, but that atually worked for me, please someone confirm. Enjoy!

**Comment by michael-shur** [comment]
the support here is made in china. look at what's going on with controlnet...

**Comment by Mofo24** [comment]
It is incredibly disruptive to my workflow practice, I need to leave generate forever running to produce a folder full of images which I then return to later to curate, Now I have to attend to the UI with  window open all the time and cant use that time to do other tasks , Im using Firefox but from the comments its a problem in most browsers so that probably not the issue.

**Comment by nickbrickett** [comment]
This issue still exists. Generate forever pauses next image run until focus returns to the Forge tab.
Fully updated & running latest version of Chrome on Windows 11 24H2.

**Comment by nickbrickett** [comment]
I've found a workaround that seems to work for the most part. I've installed this Chrome extension, which keeps the Forge tab 'alive' and prevents it being put the sleep/stopping the Generate Forever script:

**Comment by cokimax2** [comment]
> I've found a workaround that seems to work for the most part. I've installed this Chrome extension, which keeps the Forge tab 'alive' and prevents it being put the sleep/stopping the Generate Forever script: https://chromewebstore.google.com/detail/always-active-window-alwa/ehllkhjndgnlokhomdlhgbineffifcbj

I checked it works. After the install of that extension, even though the Forge UI browser is unfocused, Forge keeps processing image generation. However, the browser is minimized, Forge stops processing. So, as long as Forge browser window is open in the desktop screen, Forge keeps processing.

nicvkbrickett found an alternative solution for this problem. Good job.

**Comment by erig312** [comment]
I believe this issue is with a "[window.requestAnimationFrame()](https://developer.mozilla.org/en-US/docs/Web/API/Window/requestAnimationFrame)" calls in "http://127.0.0.1:7860/js/app/src/init.ts)". According to documentation - "calls are paused in most browsers when running in background tabs or hidden".

**Comment by Pkhtjim** [comment]
> I've found a workaround that seems to work for the most part. I've installed this Chrome extension, which keeps the Forge tab 'alive' and prevents it being put to sleep/stopping the Generate Forever script: https://chromewebstore.google.com/detail/always-active-window-alwa/ehllkhjndgnlokhomdlhgbineffifcbj

If the extension was working before, it's not working now on Opera. Just going on another browser to check emails stop Generate Forever unless I leave it open. 

Issue persists when Windows goes idle. Left on generating overnight to do some upscaling, and progress stopped not long after leaving my desk despite not having any sleep settings on the PC and the web browser with Forge left on the only tab.

**Comment by sledgehkg** [comment]
I'm guessing no resolve for this? It's just started happening to me.

**Comment by hurorida** [comment]
"--disable-features=CalculateNativeWinOcclusion" works . (Chrome 134)

**Comment by RalFingerLP** [comment]
> "--disable-features=CalculateNativeWinOcclusion" works . (Chrome 134)

what does that mean?

**Comment by ride5k** [comment]
> > "--disable-features=CalculateNativeWinOcclusion" works . (Chrome 134)
> 
> what does that mean?

https://www.reddit.com/r/chrome/comments/rxoduk/comment/hwk00hw/

**Comment by Blueblizard** [comment]
Any progress with this?

**Comment by hirorohi03** [comment]
I've only tested this on Forge Neo, but I think it will probably work on Forge as well.
https://github.com/hirorohi03/repeat-generate-4NEO

**Comment by Blueblizard** [comment]
Thanks, it works, but there's a problem with the preview images. It's not critical, but it's there. Before, each time I created an image, the created image would remain fixed until I started creating the new one. Now, the same image always remains fixed in the preview before each new image is created. Another thing I've noticed is that the image creation data below the preview isn't updating either.

**Comment by hirorohi03** [comment]
@Blueblizard 

Thanks for the report. I’ve also observed this behavior in my environment, where a preview of the first image appears briefly the moment image generation completes.
I haven’t investigated the cause in detail, but I believe this is an issue with Forge or Forge Neo itself, as it occurs frequently even when generating images in batches by increasing the `Batch count` or `Batch size` without using `repeat-generate-4NEO`.
If you can isolate this as an issue specific to `repeat-generate-4NEO`, I’ll look into it.
